Varnish Cache with auto invalidation for ES

Fifciu opened the pull request:
Hello.
I've just prepared a Varnish configuration with autoinvalidating ES. It is able to really improve your page's speed. 

Request to the:
```
/api/catalog/vue_storefront_catalog_eu/product/_search
```
Without Varnish (132ms): 
![image](https://user-images.githubusercontent.com/30155292/74104362-51353900-4b54-11ea-9aec-cbe026768750.png)
With Varnish (79ms):
![image](https://user-images.githubusercontent.com/30155292/74104376-6b6f1700-4b54-11ea-8616-d38b289eb987.png)

Elasticsearch is pretty good at caching. However, let's look at extensions that fetch some data from a real backend (M2). Example of my extension that fetches menu without Varnish (791ms):
![image](https://user-images.githubusercontent.com/30155292/74104430-cd2f8100-4b54-11ea-8051-62872c8c7e22.png)

As you see TTFB is really huge. It is because API sends the request to Magento. So normally it goes through VSF-API -> Magento. When we use Varnish we cache it in front of the VSF-API so the response is... a bit faster! (84ms)
![image](https://user-images.githubusercontent.com/30155292/74104474-2697b000-4b55-11ea-8692-4d226188ae67.png)

I am also wondering about caching `Pull Cart request` and invalidate it with `Update & Delete`. I have to check this idea.

Inside my PR, we have file `docker/varnish/README.md` there you can find every information I've prepared for developers.
https://github.com/DivanteLtd/vue-storefront-api/blob/342fe2b89ad895578cb2d9fdfdb62cffa20f6231/docker/varnish/README.md

As we have a docker image, we can easily test it locally. We just need to create the same network for each docker-compose. Usually, I run:
```sh
docker-compose -f docker-compose.nodejs.yml -f docker-compose.yml up
```

Then, in another terminal I run:
```sh
docker-compose -f docker-compose.varnish.yml up
```

So when I am experimenting with Varnish's config, I have only to rerun Varnish.
Error or tagless requests should not be cached.

andrzejewsky commented on docker-compose.yml:
```diff
@@ -17,6 +17,8 @@ services:
       - cluster.name=docker-cluster
       - bootstrap.memory_lock=true
       - "ES_JAVA_OPTS=-Xmx512m -Xms512m"
+    networks:
```
I guess this network is for varnish, but we can't enable it by default. I've seen that you created a tutorial for that which is really cool. So please remove all of the references form our docker-compose files and keep only the docker files and docker-compose.varnish.

andrzejewsky commented on src/api/catalog.js:
```diff
@@ -133,6 +140,13 @@ export default ({config, db}) => function (req, res, body) {
+        // Set output tags in response,
+        // So we could BAN them then easily
+        const tagsHeader = output.tags.join(' ')
+        res.setHeader('X-VS-Cache-Tag', tagsHeader)
```
same here... maybe config option would be good here? eg. `setVarnishCacheTags`

Fifciu commented:
I've rebased it to the new develop. Removed `network` stuff from docker composes and put configured ones in docker/varnish/docker-compose. Varnish will only work if config.varnish.enabled equals true
